template/registry_testfunc.go

```diff
@@ -25,6 +28,12 @@ func RegistryTestFunc(name string) func(*ResolvedValue, Options, language.Tag) (
 			return errorf("%w", mf2.ErrBadOperand)
 		}
 
+		// "If arg is the resolved value of an expression with a :test:function, :test:select, or :test:format
+		// annotation for which resolution has succeeded, then [..]" merge all options from resolved value.
+		if slices.Contains([]string{":test:function", ":test:format", ":test:select"}, operand.function) {
+			options = MergeOptions(operand.options, options)
```

Comment:
I wonder if this should be the responsibility of `Template`. The primary benefit is that the registry function is slimmer. And adding a new registry function is easier.

```suggestion
			options = maps.Copy(operand.options, options)
```

Reply:
Also, it enforces common behaviour across all registry functions.

Reply:
> Each function specification MAY have its own rules to preserve some options in the returned structure and discard others. In instances where a function specification does not determine whether an option is preserved or discarded, each function implementation of that specification MAY have its own rules to preserve some options in the returned structure and discard others.

So not a strong requirement, up to us what we do to options across different expressions.